File: BraintreeCore/src/main/java/com/braintreepayments/api/ConfigurationCallback.java
package com.braintreepayments.api;

import androidx.annotation.Nullable;

/**
 * Callback for receiving result of {@link BraintreeClient#getConfiguration(ConfigurationCallback)}.
 */
public interface ConfigurationCallback {

    /**
     * @param configuration {@link Configuration}
     * @param error an exception that occurred while fetching configuration
     */
    void onResult(@Nullable Configuration configuration, @Nullable Exception error);
}

File: BraintreeCore/src/main/java/com/braintreepayments/api/ConfigurationCallback.kt
package com.braintreepayments.api

/**
 * Callback for receiving result of [BraintreeClient.getConfiguration].
 */
fun interface ConfigurationCallback {
    /**
     * @param configuration [Configuration]
     * @param error an exception that occurred while fetching configuration
     */
    fun onResult(configuration: Configuration?, error: Exception?)
}
